packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -821,6 +822,81 @@ void main() {
       binding.postTest();
     });
   });
+
+  group('Accessibility announcements testing API', () {
+    testWidgets('Returns last announcement', (WidgetTester tester) async {
+      await SemanticsService.announce('announcement 1', TextDirection.ltr);
+      await SemanticsService.announce('announcement 2', TextDirection.rtl,
+          assertiveness: Assertiveness.assertive);
+
+      final CapturedAccessibilityAnnouncement last = tester.getLastAnnouncement()!;
+      expect('announcement 2', equals(last.message));
+      expect(TextDirection.rtl, equals(last.textDirection));
+      expect(Assertiveness.assertive, equals(last.assertiveness));
```

Comment:
While this doesn't affect passing or failing the test, the test API is designed for the actual value to be the first argument, and the expected value to be the second, i.e. `expect(last.assertiveness, Assertiveness.assertive)` (also `equals` is optional, because it's the default matcher anyway). The difference is that when the test fails, the error message will talk about the "expected value" and the "actual value", and it is confusing if the values are swapped.

(applies throughout the test)

Reply:
Thanks for bringing this to my attention. I totally missed it before.